logstash-core-event-java/src/main/java/com/logstash/Event.java

```diff
@@ -162,12 +162,12 @@ public String toJson()
     public static Event[] fromJson(String json)
             throws IOException
     {
-        Event[] result;
-
+        // empty/blank json string does not generate an event
         if (json == null || json.trim().isEmpty()) {
-            return new Event[]{ new Event() };
+            return new Event[]{ };
```

Comment:
because it makes one less return state to check. empty array will bubble up seamlessly to an empty Ruby array. returning null would require checking for null and acting on it.